exercism/rna-transcription/rna-transcription.js

```diff
@@ -0,0 +1,28 @@
+var DnaTranscriber = function() {};
+
+DnaTranscriber.prototype.toRna = function(dna) {
+  rna = [];
+  function UserException(message){
+    this.message = message;
+    this.name = 'UserException';
+  }
+  if(dna.match(/[^gcta]/ig)){
+    throw new UserException('Invalid input');
+  }else {
+    for (var i = 0; i<dna.length; i++){
+      if(dna[i] == 'G'){
+        rna.push('C');
+      } else if (dna[i] == 'C'){
+        rna.push('G');
+      } else if (dna[i] == 'T'){
+        rna.push('A');
+      } else if (dna[i] == 'A'){
+        rna.push('U');
+      }
+    }
+    var rnaString = rna.join('');
+    return rnaString;
+  }
+}
```

Comment:
This is really well-written! You might be able to refactor this function into a generator function. If you haven't written one before, I'd highly recommend reading about them. They would eliminate the need for declaring an empty array to push to.